File: implementations/react/src/patterns/behavioral/visitor/element-electronics.ts
import type { Visitor } from "./visitor";
import type { Element } from "./element";


/**
 *  Concrete Element que representa un artículo electrónico
 */

export class Electronics implements Element {
  name: string;
  unitPrice: number;
  qty: number;
  fragile: boolean;

  constructor(
    name: string,
    unitPrice: number,
    qty: number,
    fragile = true
  ) {
    this.name = name;
    this.unitPrice = unitPrice;
    this.qty = qty;
    this.fragile = fragile;
  }
  accept(v: Visitor) { v.visitElectronics(this); }
}

File: implementations/flutter/design_patterns_flutter/lib/patterns/behavioral/visitor/element_electronics.dart
import 'element.dart';
import 'visitor.dart';

class Electronics implements Element {
  final String name;
  final double unitPrice;
  int qty;
  final bool fragile;

  Electronics(this.name, this.unitPrice, this.qty, [this.fragile = true]);

  @override
  void accept(Visitor v) => v.visitElectronics(this);
}
